presto-native-execution/src/test/java/com/facebook/presto/nativeworker/AbstractTestNativeGeneralQueries.java

```diff
@@ -1635,6 +1635,98 @@ public void testColumnFilter()
         assertQuery(session, format("SELECT c_boolean, c_bigint, c_double, c_varchar, c_varbinary FROM %s WHERE c_varbinary = to_ieee754_64(1)", tmpTableName));
     }
 
+    @Test
+    public void testUuid()
+    {
+
+        // Valid UUIDs. Note: These evaluate on the coordinator. They are used in subsequent SQL.
+        assertQuery("SELECT cast('33355449-2c7d-43d7-967a-f53cd23215ad' AS uuid)");
+        assertQuery("SELECT cast('eed9f812-4b0c-472f-8a10-4ae7bff79a47' AS uuid)");
+        assertQuery("SELECT cast('f768f36d-4f09-4da7-a298-3564d8f3c986' AS uuid)");
+        String tmpTableName = generateRandomTableName();
+        getQueryRunner().execute(format("CREATE TABLE %s " +
+                "AS " +
+                "SELECT c_uuid  " +
+                "FROM ( " +
+                "  VALUES " +
+                "    (null), " +
+                "    ('33355449-2c7d-43d7-967a-f53cd23215ad')," +
+                "    ('eed9f812-4b0c-472f-8a10-4ae7bff79a47')," +
+                "    ('f768f36d-4f09-4da7-a298-3564d8f3c986')," +
+                "    (cast(uuid() AS VARCHAR))" +
+                ") AS x (c_uuid)", tmpTableName));
+        // Intermediate values for UUID are different between Native engine and Java (uses java.util.UUID), so these results do not match.
+        // Hence, using assertQuerySucceeds instead of assertQuery.
+        assertQuerySucceeds(format("SELECT CAST(c_uuid AS uuid) FROM %s", tmpTableName));
```

Comment:
@mbasmanova : This SQL has different results for java and native. At the outset it seems okay since the underlying physical types of UUID are different between java and native. But it might be an issue.

wdyt ?

Reply:
@aditi-pandit Curious, what are the differences? Would be nice to investigate to see if these can be removed.

Reply:
@mbasmanova : Created https://github.com/prestodb/presto/issues/23311